### ISSUE TITLE
Feature: Generate way for SitePrism to regenerate #root_element (Fixes Stale Element)

### ISSUE THREAD

**Issue by luke-hill** [site-prism/site_prism#85]
# Expected Behavior
When using a section which becomes stale, when trying to access any method or locator inside it, SitePrism should firstly try to regenerate the root_element and then if not, provide some API for doing so, to mitigate un-wanted StaleElements being caused by headers/footers being altered.

## Actual Behavior
When using a section which becomes stale, when trying to access any method or locator inside it, a StaleElement is thrown

## Steps to reproduce
N/A

**Comment by tchiaspko** [comment]
Is there a workaround?

**Comment by luke-hill** [comment]
Yes there is, but it is cumbersome (Essentially this logic is kind of what will exist when I get around to coding it).

So there is a method on stale capybara elements (You can tell their stale by a debugger inspecting them returning `Obsolete::Capybara::Element` as it's `element.inspect` invocation.

When you detect that, you can rerun `#reload`.

However the bigger issue is that something isn't always stale because **it** is stale, inside SitePrism. So the bigger issue here is to provide something more robust.

Rest assured it will be fixed. But given there is no active contributions yet into the recently set up [OpenCollective](http://www.opencollective.com/site-prism) account, it's tough to find time to work on this project at the moment.

**Comment by luke-hill** [comment]
@tchiaspko there is a partial fix available in a subsidiary gem now.

See: https://github.com/site-prism/automation_helpers/blob/main/lib/automation_helpers/extensions/capybara/node/element.rb#L23-L28

This will tell you if the element is stale. We now need to integrate this in and find a way of finding whether a section is stale (or the element), and then re-generating.

**Comment by andrewhavens** [comment]
I have had some success by using the [`allow_reload` option](https://github.com/teamcapybara/capybara/blob/master/lib/capybara/node/finders.rb#L246).
```ruby
page.my_sections(allow_reload: true).first.my_elements
```

**Comment by luke-hill** [comment]
This is interesting because I spoke to thomas a while ago about this when I was making the `#stale?` check and he indicated ways to fix it on the testing end. Not at capybara end. 

This "might" make this a whole lot easier now... Watch this space.

**Comment by gap777** [comment]
@luke-hill any update? We're injecting `allow_reload: true` into the Capybara finders via `section` and `element` declaration options... but finding (like this report) that it is insufficient, and we also need to annotate every `element` within a `reloadable` section!

**Comment by luke-hill** [comment]
Hi @gap777 - This is something I want to work on but given there isn't really any funding coming into SitePrism it's hard to prioritise it.

There is some work I need to do on making the funding streams more visible / accessible, but again it's something I've not had the time to do.

I know roughly what to do here, just not had the time sorry. I will try and get to this in 2025. If there is more demand - i.e. more comments or thumbs up here then I can try prioritise it
